### PR TITLE
[14.0][MIG] website_sale_checkout_country_vat: Migration to 14.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT32146

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]
Code review

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* Copyright 2016-2017 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define("website_sale_checkout_country_vat.dropdown", function (require) {
+    "use strict";
+    var animation = require("website.content.snippets.animation");
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]:
```suggestion
    const publicWidget = require('web.public.widget');
```

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    var CheckoutCountryVatDropdown = animation.Class.extend({
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]:
```suggestion
    publicWidget.registry.CheckoutCountryVatDropdown = publicWidget.Widget.extend({
```

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        selector: ".oe_website_sale:has(.js_country_dropdown, select[name=country_id])",
+        start: function (editable_mode) {
+            var result = this._super(editable_mode);
+            this.$address_country = this.$("select[name=country_id]");
+            this.$vat_no_country_field = this.$("#no_country_field");
+            this.$address_country.change($.proxy(this.change_address_country, this));
+            return result;
+        },
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]:
```suggestion
        start: function () {
            const result = this._super(...arguments);
            this.$address_country = this.$("select[name=country_id]");
            this.$vat_no_country_field = this.$("#no_country_field");
            return result;
        },
        
        
```

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        // Get a country element inside the vat dropdown
+        get_vat_country_selector: function (country_code) {
+            return this.$(
+                ".js_select_country_code[data-country_id='" + country_code + "']"
+            );
+        },
+    });
+
+    animation.registry.website_sale_checkout_country_vat = CheckoutCountryVatDropdown;
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]:
```suggestion
```

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+/* Copyright 2016-2017 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define("website_sale_checkout_country_vat.dropdown", function (require) {
+    "use strict";
+    var animation = require("website.content.snippets.animation");
+
+    var CheckoutCountryVatDropdown = animation.Class.extend({
+        selector: ".oe_website_sale:has(.js_country_dropdown, select[name=country_id])",
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js:8]:
```suggestion
        selector: ".oe_website_sale:has(.js_country_dropdown, select[name=country_id])",
        events: {
            'change select[name=country_id]': '_onChangeAddressCountry',
        },
```

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        start: function (editable_mode) {
+            var result = this._super(editable_mode);
+            this.$address_country = this.$("select[name=country_id]");
+            this.$vat_no_country_field = this.$("#no_country_field");
+            this.$address_country.change($.proxy(this.change_address_country, this));
+            return result;
+        },
+
+        // Change VAT flag when address country changes
+        change_address_country: function () {
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]:
```suggestion
        /**
         * Change VAT flag when address country changes
         * @private
         */
        _onChangeAddressCountry: function () {
```

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            if (this.$address_country.val() && !this.$vat_no_country_field.val()) {
+                this.get_vat_country_selector(this.$address_country.val()).click();
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]:
```suggestion
                this._getCountryOption(this.$address_country.val()).click();
```

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+            }
+        },
+
+        // Get a country element inside the vat dropdown
+        get_vat_country_selector: function (country_code) {
+            return this.$(
+                ".js_select_country_code[data-country_id='" + country_code + "']"
+            );
+        },
```
Comment (Tardo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]:
```suggestion
        /**
         * Get a country element inside the vat dropdown
         *
         * @private 
         * @param {String} country_code
         * @returns {jQueryElement}
         */
        _getCountryOption: function (country_code) {
            return this.$(
                `.js_select_country_code[data-country_id='${country_code}']`
            );
        },
```

**Review by Tardo — CHANGES_REQUESTED** [review]
You should now be able to use "publicWidget".

**Comment by ernestotejeda** [comment]
thanks @Tardo , changes done

**Comment by ernestotejeda** [comment]
ping @Tardo

File: website_sale_checkout_country_vat/views/templates.xml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+                    t-set="default_value"
+                    t-value="'vat' in checkout and checkout['vat'] or ''"
+                />
+                <t
+                    t-set="country_id"
+                    t-value="'country_id' in checkout and checkout['country_id'] or False"
+                />
+                <t
+                    t-set="default_country"
+                    t-value="countries.search([('code', '=', default_value[:2])]) or countries.search([('id', '=', country_id and int(country_id) or False)])"
```
Comment (Tardo) [inline website_sale_checkout_country_vat/views/templates.xml]:
```suggestion
                    t-value="countries.search(['|', ('code', '=', default_value[:2]), ('id', '=', country_id and int(country_id) or False)], limit=1)"
```

**Review by Tardo — CHANGES_REQUESTED** [review]
:+1: Tested in a local environment. Please check the commented "inline" query to only perform one.

**Review by Tardo — APPROVED** [review]

**Pull request merged**